From dbf8f6f9bc24199ff10c35937f7290c9663321a8 Mon Sep 17 00:00:00 2001
From: corvus-callidus <108946721+corvus-callidus@users.noreply.github.com>
Date: Wed, 21 Feb 2024 20:19:55 +0000
Subject: [PATCH] Fix -Werror removal from *FLAGS

The existing sed command to remove -Werror from *FLAGS does not cover the case
where there is an argument to -Werror. This patch changes the sed command to
remove -Werror and any following non-whitespace characters from *FLAGS.

---
 configure.ac | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 93441dc..4057bbc 100755
--- a/configure.ac
+++ b/configure.ac
@@ -4787,11 +4787,11 @@ saved_user_CPPFLAGS="$CPPFLAGS"

 ## Strip out -Werror from CFLAGS since that can cause checks to fail when
 ## compiling the test program fails due to warnings
-CFLAGS="`echo $CFLAGS | sed -e 's/-Werror//g'`"
-CXXFLAGS="`echo $CXXFLAGS | sed -e 's/-Werror//g'`"
-FCFLAGS="`echo $FCFLAGS | sed -e 's/-Werror//g'`"
-JAVACFLAGS="`echo $JAVACFLAGS | sed -e 's/-Werror//g'`"
-CPPFLAGS="`echo $CPPFLAGS | sed -e 's/-Werror//g'`"
+CFLAGS="`echo $CFLAGS | sed -e 's/-Werror\([^[:space:]]*\)//g'`"
+CXXFLAGS="`echo $CXXFLAGS | sed -e 's/-Werror\([^[:space:]]*\)//g'`"
+FCFLAGS="`echo $FCFLAGS | sed -e 's/-Werror\([^[:space:]]*\)//g'`"
+JAVACFLAGS="`echo $JAVACFLAGS | sed -e 's/-Werror\([^[:space:]]*\)//g'`"
+CPPFLAGS="`echo $CPPFLAGS | sed -e 's/-Werror\([^[:space:]]*\)//g'`"

 ## Support F9X variable to define Fortran compiler if FC variable is
 ## not used.  This should be deprecated in the future.
--
2.42.0
